services/ark-evaluator/src/evaluator/oss_providers/ragas/ragas_evaluator.py

```
@@ -0,0 +1,497 @@
+                ]
+                for alt_name in alternative_names:
+                    if alt_name in result_dict:
+                        value = result_dict[alt_name]
+                        found_field = alt_name
+                        logger.debug(f"Found {metric} using alternative name: {alt_name}")
+                        break
+
+            # Handle special cases and aliases
+            elif metric == 'helpfulness' and 'answer_relevancy' in result_dict:
+                value = result_dict['answer_relevancy']
+                found_field = 'answer_relevancy'
+            elif metric == 'clarity' and 'answer_similarity' in result_dict:
+                value = result_dict['answer_similarity']
+                found_field = 'answer_similarity'
+
+            # Last resort: try exact metric name
+            elif metric in result_dict:
+                value = result_dict[metric]
+                found_field = metric
+                logger.debug(f"Found {metric} using exact name match")
+            
+            # Process the value
+            if value is not None:
+                # Handle NaN values that can occur in RAGAS evaluation
+                if math.isnan(float(value)):
+                    logger.warning(f"RAGAS returned NaN for metric {metric} (field: {found_field}), using fallback score")
+                    scores[metric] = 0.7  # Reasonable fallback for NaN
+                else:
+                    scores[metric] = float(value)
+                    logger.debug(f"Successfully extracted {metric} = {value} (from field: {found_field})")
+            else:
+                available_fields = list(result_dict.keys())
+                logger.warning(f"No RAGAS result found for metric: {metric}, using default. Available fields: {available_fields}")
+                scores[metric] = 0.5
+        
+        logger.info(f"Extracted scores: {scores}")
+        return scores
+    
+    @staticmethod
+    def get_fallback_scores(
```

should we raise error instead of fallback ?